tests/integration-tests/tests/budgets/test_budgets.py

```diff
@@ -33,6 +37,49 @@ def test_update_budget_properties(pcluster_config_reader, clusters_factory, test
     # Add one budget of each BudgetCategory, update them, and check that the fields match the expected values
     _test_update_cluster_and_check_fields(cluster, budget_client, account_id, region, pcluster_config_reader)
 
+    # Enable fleet stop action, trigger sns topic notification, and assert that the fleet has stopped
+    _test_stop_compute_fleet(cluster, pcluster_config_reader)
+
+
+def _test_stop_compute_fleet(cluster, pcluster_config_reader):
+    # Create the initial cluster with a budget triggered compute-fleet stop action
+    stop_fleet_config_file = pcluster_config_reader(config_file="budget_with_fleet_stop.yaml")
+    cluster.update(str(stop_fleet_config_file), force_update="true")
+    time.sleep(30)
+
+    cluster_info = cluster.describe_cluster()
+
+    assert_that(cluster_info.get("computeFleetStatus")).is_in("RUNNING", "ENABLED")
+
+    sns_client = boto3.client("sns")
+
+    # Look for the right topic using boto3
+    topic_list = sns_client.list_topics().get("Topics")
+    budget_topic_arn = None
+    for topic in topic_list:
+        topic_arn = topic.get("TopicArn")
+
+        if topic_arn is not None and cluster.name in topic_arn:
+            topic_attributes = sns_client.get_topic_attributes(TopicArn=topic_arn).get("Attributes")
+
+            display_name = topic_attributes.get("DisplayName")
+            if display_name is not None and "BudgetTriggeredNotification" in display_name:
+                budget_topic_arn = topic_arn
+                break
+
+    assert_that(budget_topic_arn).is_not_none()
+
+    # Message should trigger fleet stopper lambda
+    sns_client.publish(TopicArn=budget_topic_arn, Message="Test Stop Fleet")
+
+    _assert_compute_fleet_stop(cluster)
+
+
+@retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))
+def _assert_compute_fleet_stop(cluster):
+    cluster_info = cluster.describe_cluster()
+    assert_that(cluster_info.get("computeFleetStatus")).is_in("STOP_REQUESTED", "STOPPED", "DISABLED", "STOPPING")
```

Comment:
since we have a retry and wait here, can we only assert they are STOPPED or DISABLED?

Reply:
Yes, that makes sense. Updated!